### ISSUE TITLE
v1.29.0 contains a breaking change to encoding/protowire.Number.IsValid

### ISSUE THREAD

**Issue by dylan-bourque** [protocolbuffers/protobuf#12218]
In v1.29.0, the `IsValid()` method on `encoding/protowire.Number` was changed to no longer return `false` for tag values between `FirstReservedNumber` and `LastReservedNumber`, inclusive.  See [this commit](https://github.com/protocolbuffers/protobuf-go/commit/31a59208116ee2cacd5b3e4f64cf17ad733e18db).  This change in exported behavior triggered test failures in my code.

When I checked the release notes for v1.29.0 there was no mention of this breaking change.

I was able to change my code because I didn't have anything that actually relied on the previous behavior (the tests were for completeness more than functional requirements) but I thought it worth calling out the lack of mention.

**Comment by jskeet** [comment]
Did you intend this issue to be in https://github.com/protocolbuffers/protobuf-go rather than https://github.com/protocolbuffers/protobuf?

**Comment by jskeet** [comment]
I don't, I'm afraid - it may even be that this repo *is* the right place. I'll contact one of the Protobuf Go team and ask them to weigh in.

**Comment by leitzler** [comment]
fwiw - issues should be reported in the old repo - https://github.com/golang/protobuf/issues as per the readme, https://github.com/protocolbuffers/protobuf-go#reporting-issues

**Comment by dylan-bourque** [comment]
> This change was necessary to allign with the protobuf language specification.

From the proto3 guide on protobuf.dev: "You also cannot use the numbers 19000 through 19999."  One could argue that it was correct before where `IsValid()` was returning `false` for tag values in that range.  I realize I'm splitting hairs a bit, but from a Semantic Versioning and API compatibility standpoint this was 100% a breaking change.

Like I said before, I was able to fix my code because the failing tests were there for completeness (those values were not allowed so I had tests that verified that constraint held) so it's not really a big deal.

**Comment by lfolger** [comment]
I see. Thanks for the feedback. I'll bring this up with the team and figure out if we can do better in the future.

**Comment by lfolger** [comment]
I discussed this with the team and we came up with the following conclusion.

You are right and if we would follow semantic versioning strictly this would require a new version.

However, we think that the costs of a new version are not worth the benefits.
There are [very few users](https://cs.github.com/?scopeName=All+repos&scope=&q=google.golang.org%2Fprotobuf%2Fencoding%2Fprotowire+lang%3Ago+%2FNumber.*%5C.IsValid%5C%28%2F) of this API.
Introducing a new major version requires users to update their code to depend on the new major version or they are stuck with the old version.
Since we do not have the capacity to maintain two version we would only maintain the newer version.

In other words, we think that the current decision is a net-positive for the Go protobuf ecosystem as whole even if it means we do not strictly adhere to semantic version.  We will follow this principle for future will submit minor breaking changes without declaring a new major version.

**Comment by dylan-bourque** [comment]
That's certainly your decision to make.  I just wanted to bring it up because it is, in fact, a breaking change and the kind of thing that the Go Modules guidance tells us should be a major version bump.  It's also a subtle breaking change because it won't cause any build failures, it will just behave differently at runtime.

At the very least it needs to be called out in the release notes.
